keanu-project/src/main/java/io/improbable/keanu/distributions/tensors/continuous/TensorSmoothUniform.java

```diff
@@ -0,0 +1,119 @@
+package io.improbable.keanu.distributions.tensors.continuous;
+
+import io.improbable.keanu.vertices.dbltensor.DoubleTensor;
+import io.improbable.keanu.vertices.dbltensor.KeanuRandom;
+
+public class TensorSmoothUniform {
+
+    private TensorSmoothUniform() {
+    }
+
+    public static DoubleTensor sample(int[] shape, DoubleTensor xMin, DoubleTensor xMax, double edgeSharpness, KeanuRandom random) {
+
+        DoubleTensor r1 = random.nextDouble(shape);
+        DoubleTensor r2 = random.nextDouble(shape);
+
+        final DoubleTensor bodyWidth = xMax.minus(xMin);
+        final DoubleTensor shoulderWidth = bodyWidth.times(edgeSharpness);
+        final DoubleTensor rScaled = r1.times(bodyWidth.plus(shoulderWidth)).plusInPlace(xMin.minus(shoulderWidth.div(2)));
+        final DoubleTensor bodyHeight = bodyHeight(shoulderWidth, bodyWidth);
+
+        final DoubleTensor firstConditional = rScaled.getGreaterThanOrEqualToMask(xMin);
+        firstConditional.timesInPlace(rScaled.getLessThanOrEqualToMask(xMax));
```

Comment:
`firstConditional = firstConditional.timesInPlace` just in case.

Reply:
nice, changed